### ISSUE TITLE
Input data

### ISSUE THREAD

**Issue by frajem** [kracwarlock/action-recognition-visual-attention#2]
Hello @kracwarlock 
I have two questions about the input data:
1. When training for Hollywood2, I get a memory error "Cannot allocate memory" before the training loop starts, probably due to the big amount of data. In the pre-processing, I sparsify the features, and my training feature file is about 10 GB. How big is this file for you? 
2. If I use a subset of the dataset (to avoid the above memory problem), the training starts fine but at Epoch 0 Update 1240 an error occurs: "NaN detected in cost". 

Thanks!

**Comment by kracwarlock** [comment]
Hi @frajem 
1. My training set HDF5 file is 6 GB, validation is 2.7 GB.
   Combined, the train file is 8.6 GB and the test file is 9. 8GB.
   How much RAM and GPU memory do you have and what batch size are you using?
   As far as I remember, I had a batch size of 128 and the server ram was ~48 GB and the GPU memory was 6 GB. The data handler loads one batch into memory at once so a smaller batch size should make your troubles go away unless your RAM is very small.
2. The NaN seems very suspicious. What version of Theano are you using? Can you try upgrading it to the latest stable release or downgrade to 0.6 if already on latest stable release?
   In a separate project I had found some NaNs occurring when using CUDA and CUDNN 7.5 with Theano 0.7 but the code worked perfectly in a lower version of both. It might be a Theano bug. Let me know if the issue still remains and I will look into it.

**Comment by frajem** [comment]
Thanks @kracwarlock 
For question 2: I have Cuda 7.0 and I tried with both Theano 0.7 and 0.6, but I still get the NaN in the cost. Is it possible to get your features files for Hollywood2?

**Comment by kracwarlock** [comment]
@frajem it would be difficult to send 20 GB files since I am no longer at University of Toronto but I'll try. Can you check that your features don't contain Inf / NaN?

**Comment by kracwarlock** [comment]
@frajem Check this out: http://deeplearning.net/software/theano/library/config.html#config.config.nvcc.fastmath
If you use fastmath that can cause NaNs

**Comment by frajem** [comment]
Thanks @kracwarlock 
No, my features do not seem to contain Inf / NaN and I'm not using fastmath.
I have switched from SGD to Adam optimizer and training is now proceeding fine, without NaNs.

**Comment by kracwarlock** [comment]
@frajem That's interesting. What learning rate were you using?

**Comment by frajem** [comment]
@kracwarlock The default one for Hollywood2 in actrec_mAP.py: 0.0001.

**Comment by frajem** [comment]
@kracwarlock 
Two more questions:
1. Did you extract the features from each single frame, or did you sub-sample the frames? I'm trying to understand why I get bigger files (10 GB for the training features, 13 GB for test features).
2. If it's not possible to share the features, can you please share a python script for extracting them (e.g., for Hollywood2)?

Thanks in advance!

**Comment by kracwarlock** [comment]
@frajem Hi. I am sorry for all the delay. I was very busy with my thesis and graduation.
1. I extracted the features from each single frame after reducing them to 256x256
2. @YknZhu (Yukun Zhu in my paper) extracted the features for me using a matlab script. I am sharing that here http://www.cs.toronto.edu/~yukun/extracting_feature.m. To combine the individual files generated by this script that he sent me I used https://gist.github.com/kracwarlock/96499936487d6125dd010319669c6648
